### ISSUE TITLE
Fetch imagery weekly

### ISSUE THREAD

**Issue by arch0345** [openstreetmap/iD#9730]
### Description

Currently, iD only fetches imagery from [editor-layer-index](https://https://github.com/osmlab/editor-layer-index) before each minor release. Since the last release that updated imagery was [over four months ago](https://github.com/openstreetmap/iD/releases/tag/v2.25.0), many sources that have been added/updated since then are not available for those that use iD. It would be nice to have the imagery list updated independent of iD releases similar to [id-tagging-schema](https://github.com/openstreetmap/id-tagging-schema), [osm-community-index](https://github.com/osmlab/editor-layer-index), and [name-suggestion-index](https://github.com/osmlab/name-suggestion-index).

Since it looks like a prior attempts to version the index itself weren't successful, I propose creating a new repository that fetches imagery from ELI weekly. This repository would function in a similar way to how [name-suggestion-index](https://github.com/osmlab/name-suggestion-index) uses the [nsi-collector](https://github.com/ideditor/nsi-collector) repo to periodically collect common brands/operators from the OSM planet file.

[Here's a proof of concept I made](https://github.com/arch0345/eli-collector)

See also:
* #4994
* #7425
* #8466

Edit: Started writing this yesterday and didn't notice #9729 which seems to touch on the same issue

**Comment by bhousel** [comment]
I’ve already built this thing like twice and #7425 is the thing to switch to. 

Please PM me on Slack for the full story on what happened.

**Comment by Nekzuris** [comment]
This desperately needs to move on, imagery is now even more frequently updated than it was 4 years ago.

**Comment by gy-mate** [comment]
> Please PM me on Slack for the full story on what happened.

@bhousel Could you please give as a summary on this? Unfortunately I don't have an [osmus.slack.com](https://osmus.slack.com) account.

**Comment by gy-mate** [comment]
@arch0345 The TODO is to...

1. ask the OSMF if they like this improvement and would like to proceed with the steps below
2. transfer the ownership of https://github.com/arch0345/eli-collector to https://github.com/osmlab
3. set up a cron job that runs https://github.com/arch0345/eli-collector/blob/1ac9772d637331afcb0dfa9677b77dfce9096565/scripts/update_imagery.js and https://github.com/arch0345/eli-collector/blob/1ac9772d637331afcb0dfa9677b77dfce9096565/scripts/bump_version.js periodically (e.g. every day)
4. make references of https://github.com/openstreetmap/iD/blob/f00fa14b90f7c9dc2789a701bfa06a6db40b8065/data/imagery.json point to the same file in the transferred repo instead?

**Comment by k-yle** [comment]
or alternatively: https://github.com/osmlab/editor-layer-index/pull/2530

**Comment by gy-mate** [comment]
Nice, that looks like a better solution to me! :)

So the TODOs are to...

1. review and merge https://github.com/osmlab/editor-layer-index/pull/2530
2. make references of https://github.com/openstreetmap/iD/blob/f00fa14b90f7c9dc2789a701bfa06a6db40b8065/data/imagery.json point to the `latest` version (tag) of https://github.com/osmlab/editor-layer-index/blob/35638fef8bbfa57f76d3db1fdc46c2048d35f33d/imagery.json.

**Comment by danieldegroot2** [comment]
_Mentioned in [ID/Community Chats - 2nd_July_2025](https://wiki.openstreetmap.org/wiki/ID/Community_Chats/2025#Community_Chat_2nd_July_2025_15:00_%E2%80%93_16:00_(UTC))._

See also [osmlab/editor-layer-index#2625](https://github.com/osmlab/editor-layer-index/issues/2625) / [osmlab/editor-layer-index#538](https://github.com/osmlab/editor-layer-index/issues/538)

**Comment by tyrasd** [comment]
One additional comment: the code in iD is essentially already ready to fetch imagery data dynamically from a CDN or similar location.

**Comment by k-yle** [comment]
yeah, for reference, this is the implementation from a fork of iD which already fetches directly from ELI: fd74f9fb48f6390f68e5a90529bb119852e345bc
